Tomlet.Tests/TomlTableArrayTests.cs

```diff
@@ -49,30 +49,39 @@ public void ComplexTableArraysAreSupported()
             //Apple
             var firstFruit = Assert.IsType<TomlTable>(document.GetArray("fruits")[0]);
             Assert.Equal("apple", firstFruit.GetString("name"));
-            
+
             var physical = Assert.IsType<TomlTable>(firstFruit.GetValue("physical"));
+            var jam = Assert.IsType<TomlTable>(firstFruit.GetValue("jam"));
             var varieties = Assert.IsType<TomlArray>(firstFruit.GetValue("varieties"));
             
             Assert.Equal("red", physical.GetString("color"));
             Assert.Equal("round", physical.GetString("shape"));
-            
+
+            Assert.Equal("yellow", jam.GetString("color"));
+            Assert.Equal("sticky", jam.GetString("feel"));
+
             Assert.Equal(2, varieties.Count);
             Assert.Equal("red delicious", Assert.IsType<TomlTable>(varieties[0]).GetString("name"));
             Assert.Equal("granny smith", Assert.IsType<TomlTable>(varieties[1]).GetString("name"));
             
             //Banana
             var secondFruit = Assert.IsType<TomlTable>(document.GetArray("fruits")[1]);
             Assert.Equal("banana", secondFruit.GetString("name"));
-            
+
+            physical = Assert.IsType<TomlTable>(secondFruit.GetValue("physical"));
+            var newtonian = Assert.IsType<TomlTable>(physical.GetValue("newtonian"));
             varieties = Assert.IsType<TomlArray>(secondFruit.GetValue("varieties"));
 
+            Assert.Equal("yellow", physical.GetString("color"));
+            Assert.Equal(118, newtonian.GetInteger("weight"));
+
             Assert.Single(varieties, val => Assert.IsType<TomlTable>(val).GetString("name") == "plantain");
         }
 
         [Fact]
         public void DefiningATableArrayWithTheSameNameAsATableThrowsAnException()
         {
-            Assert.Throws<TomlTableArrayAlreadyExistsAsNonArrayException>(() => GetDocument(TestResources.DefiningAsArrayWhenAlreadyTableTestInput));
+            Assert.Throws<TomlTypeMismatchException>(() => GetDocument(TestResources.DefiningAsArrayWhenAlreadyTableTestInput));
```

Comment:
Assuming you update the exception type in the other comment, you'll need to update the expected exception type here.